devito/operations/interpolators.py

```diff
@@ -241,10 +291,13 @@ def _interp_idx(self, variables, implicit_dims=None, pos_only=()):
         temps.extend(self._coeff_temps(implicit_dims))
 
         # Substitution mapper for variables
-        mapper = self._rdim.getters
-        idx_subs = {v: v.subs({k: c + p
-                    for ((k, c), p) in zip(mapper.items(), pos)})
-                    for v in variables}
+        mapper = self._rdim(subdomain=subdomain).getters
+
+        # Index substitution to make in variables
+        subs = {k: c + p for ((k, c), p) in zip(mapper.items(), pos)}
```

Comment:
subs = {ki: c + p for ((k, c), p) in zip(mapper.items(), pos)  for ki in {k, k.root}}